corehq/apps/reports/views.py

```diff
@@ -1358,6 +1358,23 @@ def download_form(request, domain, instance_id):
     return couchforms_views.download_form(request, instance_id)
 
 
+def _form_instance_to_context_url(domain, instance):
+    try:
+        build = Application.get(instance.build_id)
+    except ResourceNotFound:
+        raise Http404(_('Application not found.'))
+
+    for module in build.get_modules():
```

Comment:
You can use `get_form_by_xmlns`: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/models.py#L4863-4863